bql/planner/data_access.go

```diff
@@ -409,7 +409,12 @@ func simpleFetch(ctx context.Context, gs []storage.Graph, cls *semantic.GraphCla
 // table. The semantic graph clause is also passed to be able to identify what
 // bindings to set.
 func addTriples(ts <-chan *triple.Triple, cls *semantic.GraphClause, tbl *table.Table) error {
+	var lastErr error
 	for t := range ts {
+		if lastErr != nil {
+			// Drain the channel to avoid leaking goroutines.
+			continue
+		}
```

Comment:
I think we can simplify this by always iterating on the ts channel after the loop which would effectively drain the channel.
It would be then your option of ignoring the error altogether or checking if it is non nil and breaking the loop to the draining one. Maybe even extract the channel draining logic to a DrainChannel function.

Specifically I think a break, instead of a continue will make readability easier.

Reply:
This change is in fact added with the PR #119 ("Fix bug of TYPE bindings with non-node objects and AT bindings with immutable triples"), appearing here because of the rebase.
I transferred this comment to the conversation there on the PR #119 there then.